cloud/blockstore/tools/csi_driver/internal/driver/node_test.go

```diff
@@ -411,3 +412,111 @@ func TestPublishUnpublishDeviceForInfrakuber(t *testing.T) {
 	})
 	require.NoError(t, err)
 }
+
+func TestGetVolumeStatCapabilitiesInVmMode(t *testing.T) {
+	tempDir := os.TempDir()
+
+	nbsClient := mocks.NewNbsClientMock()
+	mounter := mounter.NewMock()
+
+	podID := "test-pod-id-13"
+	diskID := "test-disk-id-42"
+	socketsDir := filepath.Join(tempDir, "sockets")
+	targetPath := filepath.Join(tempDir, "pods", podID, "volumes", diskID, "mount")
+	targetFsPathPattern := filepath.Join(tempDir, "pods/([a-z0-9-]+)/volumes/([a-z0-9-]+)/mount")
+
+	nodeService := newNodeService(
+		"testNodeId",
+		"testClientId",
+		true, // vmMode
+		socketsDir,
+		targetFsPathPattern,
+		"", // targetBlkPathPattern
+		nbsClient,
+		nil, // nfsClient
+		mounter,
+	)
+
+	ctx := context.Background()
+	resp, err := nodeService.NodeGetCapabilities(
+		ctx,
+		&csi.NodeGetCapabilitiesRequest{})
+	assert.Nil(t, err)
+
+	capabilityIndex := slices.IndexFunc(resp.GetCapabilities(), func(capability *csi.NodeServiceCapability) bool {
+		rpc := capability.GetRpc()
+		if rpc == nil {
+			return false
+		}
+		return rpc.GetType() == csi.NodeServiceCapability_RPC_GET_VOLUME_STATS
+	})
+	assert.NotEqual(t, -1, capabilityIndex)
+
+	stat, err := nodeService.NodeGetVolumeStats(ctx, &csi.NodeGetVolumeStatsRequest{
+		VolumeId:   diskID,
+		VolumePath: targetPath,
+	})
+	require.NoError(t, err)
+	assert.Equal(t, 2, len(stat.GetUsage()))
+
+	bytesUsage := stat.GetUsage()[0]
+	assert.Equal(t, bytesUsage.Unit, csi.VolumeUsage_BYTES)
+	assert.Equal(t, bytesUsage.Used+bytesUsage.Available, bytesUsage.Total)
+
+	nodesUsage := stat.GetUsage()[1]
+	assert.Equal(t, nodesUsage.Unit, csi.VolumeUsage_INODES)
+	assert.Equal(t, nodesUsage.Used+nodesUsage.Available, nodesUsage.Total)
```

Comment:
It seems that you are just checking for the invariant. It would be nice to create some files and then inspect actual inodes count and sizes.

Reply:
I'm not sure that it's possible because in fact we don't mount volume in this test and overall numbers of inodes/bytes depend on the state of vm filesystem. 

By the way I added another assert to check that total number of inodes/bytes are not zero.